src/node.zig

```diff
@@ -331,55 +465,147 @@ fn request(env: c.napi_env, info: c.napi_callback_info) callconv(.C) c.napi_valu
         translate.throw(env, "Unknown operation.") catch return null;
     }
 
+    const message = context.client.get_message() orelse {
+        translate.throw(
+            env,
+            "Message pool has been exhausted.",
+        ) catch return null;
+    };
+    defer context.client.unref(message);
+
     const operation = @intToEnum(Operation, @intCast(u8, operation_int));
-    const events = decode_slice_from_array(env, argv[2], operation) catch |err| switch (err) {
+    const body_length = decode_events(
+        env,
+        argv[2],
+        operation,
+        message.buffer[@sizeOf(Header)..],
+    ) catch |err| switch (err) {
         error.ExceptionThrown => {
             return null;
         },
```

Comment:
We could make this a one-liner without using a block to enclose the `return null,`